test/gradcheck.jl

```diff
@@ -83,6 +83,7 @@ end
 
 @test gradtest(x -> prod(x, dims = (2, 3)), (3,4,5))
 @test gradtest(x -> prod(x), (3,4))
+@test gradient(x -> prod((1,2,3)), 1)[1] === nothing 
```

Comment:
The test should check for a valid gradient result

```suggestion
x = (1,2,3)
@test gradient(x -> prod(x), x)[1] == (6,3,2)
```